src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/FormatFilterExtension.java

```diff
@@ -77,12 +81,14 @@ public Object call(final RenderContext renderContext, Object... arguments) {
         if (hasPlaceHolders) {
             return getFormattedString(runtimeObjectModel, source, formatObject);
         }
+
         try {
-            // somebody will hate me for this
-            new SimpleDateFormat(source);
+            // try to parse as DateTimeFormatter
```

Comment:
I am not sure how this branch can be reached, if it is dead code we should remove it.

Reply:
@raducotescu Can you answer this?

Reply:
I'm afraid I don't understand the question. The code before attempted to check if the passed string was a valid date format pattern, which is similar to what the new code does.

Reply:
The question is rather under which circumstances do you end in this area as there are 4 return statements before this line (which IMHO cover all configuration options...)

Reply:
Ah, that's simple - you have no `type` option passed along the `format` option, so we have to figure out what kind of pattern the user provided.

Reply:
@Buuhuu Can you make sure this is covered by your unit test?

Reply:
But only for the condition that the `format` option is not one of `Date` or `Calendar`, in which case the `getDateFormattedString()` always returns null https://github.com/apache/sling-org-apache-sling-scripting-sightly/pull/11/commits/7e43ff55a4d27c89a35fa0041f9b97534881e093 

I cannot find anything in the spec about this behaviour but it seems reasonable.